### ISSUE TITLE
Ability to specify a proxy for each driver

### ISSUE THREAD

**Issue by Cruikshanks** [DEFRA/quke#46]
We have found that in some build environments access to the internet is provided by a proxy. Therefore for **Quke** to work we need to be able to tell **PhantomJS** and **Selenium** what the address and port for the proxy server is.

**Comment by Cruikshanks** [comment]
Have found some examples of how to do this

http://stackoverflow.com/a/6595850/6117745
```ruby
Capybara.register_driver :selenium do |app|
    profile = Selenium::WebDriver::Firefox::Profile.new

    profile["network.proxy.type"] = 1 # manual proxy config
    profile["network.proxy.http"] = "http://example.com"
    profile["network.proxy.http_port"] = 80

    Capybara::Selenium::Driver.new(app, :profile => profile)
end
```

https://gist.github.com/antonyh/5338945b37b6b52a98f5
```ruby
Capybara.register_driver :poltergeist_proxy do |app|                                                                                                                                                                                        
    options = {                                                                                                                                                                                                                               
      phantomjs_options: [                                                                                                                                                                                                                    
        '--ignore-ssl-errors=yes',                                                                                                                                                                                                            
        "--proxy=#{CapybaraProxy.proxy.host}:#{CapybaraProxy.proxy.port}"                                                                                                                                                                     
      ]                                                                                                                                                                                                                                       
    }                                                                                                                                                                                                                                         
    Capybara::Poltergeist::Driver.new(app, options)
end

Capybara.register_driver :webkit_proxy do |app|                                                                                                                                                                                             
    driver = Capybara::Driver::Webkit.new(app)                                                                                                                                                                                                
    driver.browser.set_proxy(:host => CapybaraProxy.proxy.host,                                                                                                                                                                               
                             :port => CapybaraProxy.proxy.port)                                                                                                                                                                               
    driver.browser.ignore_ssl_errors                                                                                                                                                                                                          
    driver
end

Capybara.register_driver :selenium_proxy do |app|                                                                                                                                                                                           
    profile = Selenium::WebDriver::Firefox::Profile.new                                                                                                                                                                                       
    profile.proxy = Selenium::WebDriver::Proxy.new(                                                                                                                                                                                           
      :http => "#{CapybaraProxy.proxy.host}:#{CapybaraProxy.proxy.port}",                                                                                                                                                                     
      :ssl => "#{CapybaraProxy.proxy.host}:#{CapybaraProxy.proxy.port}")                                                                                                                                                                      
    Capybara::Selenium::Driver.new(app, :profile => profile)
end 
```

http://www.rubydoc.info/gems/selenium-webdriver/0.0.28/Selenium%2FWebDriver%2FFirefox%2FProfile%3Ainitialize
```ruby
profile = Selenium::WebDriver::Firefox::Profile.new
profile['network.proxy.http'] = 'localhost'
profile['network.proxy.http_port'] = 9090

driver = Selenium::WebDriver.for :firefox, :profile => profile
```

**Comment by Cruikshanks** [comment]
For future reference when it comes to figuring out what options you can pass to chrome it essentially appears to be this list http://peter.sh/experiments/chromium-command-line-switches

You can pass them in as an array with the switches tag for example

```ruby
Capybara::Selenium::Driver.new(
    app,
    browser: :chrome,
    switches: ["--proxy-server=localhost:8080"]
)
```

Got this inspiration when I found this https://github.com/SeleniumHQ/selenium/wiki/Ruby-Bindings#chrome

**Comment by Cruikshanks** [comment]
This was resolved #47
